### PR TITLE
TW-353: improve for the swipe

### PR DESCRIPTION
**Pull request by sherlockvn**
## Demo
https://github.com/linagora/twake-on-matrix/assets/43041967/3a7fe338-60b9-4258-b3a6-64356e27c7c1

https://github.com/linagora/twake-on-matrix/assets/43041967/4da2b688-f63f-4f48-91e7-9e99a6ee58ea

### CODE REVIEW

File: lib/pages/chat/events/message.dart [hunk 1]
```diff
@@ -406,11 +406,14 @@ class Message extends StatelessWidget {
...
               background: const Padding(
-                padding: EdgeInsets.symmetric(horizontal: 12.0),
+                padding: EdgeInsets.symmetric(horizontal: 0.0),
                 child: Center(
                   child: Icon(Icons.reply_outlined),
                 ),
               ),
+              maxOffset: 0.4,
+              movementDuration: const Duration(milliseconds: 100),
+              swipeIntensity: 2.5,
```
Comment (hoangdat) [inline lib/pages/chat/events/message.dart:416]: what is it?

File: pubspec.yaml [hunk 2]
```diff
@@ -83,7 +83,10 @@ dependencies:
   shared_preferences: 2.0.15 # Pinned because https://github.com/flutter/flutter/issues/118401
   slugify: ^2.0.0
   skeletons: ^0.0.3
-  swipe_to_action: ^0.2.0
+  swipe_to_action:
```
Comment (hoangdat) [inline pubspec.yaml:86]: how about to change it to https://pub.dev/packages/flutter_swipe_action_cell?

Comment (nqhhdev) [reply pubspec.yaml:86]: I agree , `swipe_to_action` is out of date 2 years agooo :)))))))

File: lib/pages/chat/events/message.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat/events/message.dart:418]: duration of animation movement

File: lib/pages/chat/events/message.dart [hunk 3]
```diff
@@ -406,11 +406,14 @@ class Message extends StatelessWidget {
             Swipeable(
               key: ValueKey(event.eventId),
               background: const Padding(
-                padding: EdgeInsets.symmetric(horizontal: 12.0),
+                padding: EdgeInsets.symmetric(horizontal: 0.0),
```
[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat/events/message.dart]: Don't need to change it. i forget to change it back.

File: lib/pages/chat/events/message.dart [hunk 4]
```diff
@@ -411,6 +412,10 @@ class Message extends StatelessWidget {
                   child: Icon(Icons.reply_outlined),
                 ),
               ),
+              onOverScrollTheMaxOffset:() => HapticFeedback.heavyImpact(),
+              maxOffset: 0.4,
+              movementDuration: const Duration(milliseconds: 100),
+              swipeIntensity: 2.5,
```
Comment (hoangdat) [inline lib/pages/chat/events/message.dart:418]: And this?

**Comment by hoangdat** [comment]
What is the error in the demo after you finished swipe?

**Review by hoangdat — COMMENTED** [review]
Can you paste the link of swipe lib also?

**Comment by sherlockvn** [comment]
> What is the error in the demo after you finished swipe?

It's the demo of image sending error because I'm using linagora server.

**Comment by sherlockvn** [comment]
> Can you paste the link of swipe lib also?

I have updated it in description

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
image

Please fix it

**Review by hoangdat — APPROVED** [review]

**Pull request merged**